In [1]:
import pandas as pd 

In [2]:
df= pd.read_csv('/Users/abhishekkaul/Downloads/train.csv.zip')

In [3]:
df

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[200000 rows x 202 columns]

In [4]:
df['var_0'].min()

0.4084

In [5]:
df['var_0'].max()

20.315

In [6]:
df['var_1'].min()

-15.0434

In [7]:
df['var_1'].max()

10.3768

In [8]:
df['var_2'].min()

2.1171

In [9]:
df['var_2'].max()

19.352999999999998

# compute the following probablities:


\begin{equation}
p(target=0 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))
\end{equation}
\begin{equation}
p((1 \leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) | target=0)
\end{equation}
\begin{equation}
p(target=1 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))
\end{equation}
\begin{equation}
p((1 \leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) | target=1)
\end{equation}

# lets compute the first one:

\begin{equation}
p(target=0 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))
\end{equation}


# lets say event (A) = (target = 0 ), & event (b) = (1<= VARo<=15 and -10<=VAR1<=9.5 and 4<=VAR2 <= 18)


# now according to the conditional probablity formula:

\begin{equation}
p(A|B) = \frac{p(A\cap B)}{p(B)}
\end{equation}

# Therefore:

\begin{equation}
p(target=0 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)) = \frac{p((target=0 \cap (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))}{p(1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)}
\end{equation}


In [10]:
boolean_mask = df['target'] == 0

In [11]:
df[boolean_mask]

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[179902 rows x 202 columns]

In [12]:
boolean_mask_1 = df[(df['target']==0) & (df['var_0']>= 1) & (df['var_0']<= 15) & (df['var_1'] >= -10) & (df['var_1'] <= 9.5) & (df['var_2']>= 4) & (df['var_2']<= 18)]

In [13]:
boolean_mask_1

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[163891 rows x 202 columns]

In [14]:
joint_probablity_1 = boolean_mask_1.shape[0]/df.shape[0]

In [15]:
joint_probablity_1

0.819455

In [16]:
boolean_mask_2 = df[(df['var_0']>=1) & (df['var_0']<=15) & (df['var_1']>= -10) & (df['var_1']<= 9.5) & (df['var_2']>= 4) & (df['var_2'] <= 18)]

In [17]:
boolean_mask_2

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[181161 rows x 202 columns]

In [18]:
joint_probablity_2 = boolean_mask_2.shape[0]/df.shape[0]

In [19]:
joint_probablity_2

0.905805

In [20]:
conditional_probablity_1 = joint_probablity_1/joint_probablity_2

In [21]:
conditional_probablity_1

0.9046704312738394

# so condtional probablity for the following equation is:

\begin{equation}
p(target=0 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)) = 0.904670
\end{equation}

# now lets compute the probablity for:

\begin{equation}
p((1 \leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) | target=0)
\end{equation}

# lets say event (A) = (1<= VARo <= 15 & -10<= VAR1 <= 9.5 & 4 <= VAR2 <=18) and event (B) = (target = 0)

\begin{equation}
p(A|B) = \frac{p(A\cap B)}{p(B)}
\end{equation}


\begin{equation}
p((1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) | target = 0) = \frac{p(1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)\cap target = 0)}{p(target = 0)}
\end{equation}

# again we have to compute two different probablities:
\begin{equation}
p((1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)\cap(target=0)
\end{equation}

# and

\begin{equation}
p(target = 0)
\end{equation}


In [22]:
boolean_mask_3 = df[(df['var_0']>=1) & (df['var_0']<=15) & (df['var_1']>= -10) & (df['var_1']<= 9.5) & (df['var_2']>= 4) & (df['var_2'] <= 18) & (df['target']==0)]

In [23]:
boolean_mask_3

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[163891 rows x 202 columns]

In [24]:
joint_probablity_3 = boolean_mask_3.shape[0]/ df.shape[0]

In [25]:
joint_probablity_3

0.819455

In [26]:
boolean_mask_4 = df[df['target']== 0]

In [27]:
joint_probablity_4 = boolean_mask_4.shape[0]/ df.shape[0]

In [28]:
joint_probablity_4

0.89951

In [29]:
conditional_probablity_2 = joint_probablity_3/joint_probablity_4

In [30]:
conditional_probablity_2

0.9110015452857667

# so the conditonal probablity for the following equation is :

\begin{equation}
p((1 \leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) | target=0)= 0.9110015
\end{equation}

# now lets compute the probablity for:

\begin{equation}
p(target=1 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))
\end{equation}

# lets say event (A) = (target = 1 ), & event (b) = (1<= VARo<=15 and -10<=VAR1<=9.5 and 4<=VAR2 <= 18)


\begin{equation}
p(A|B) = \frac{p(A\cap B)}{p(B)}
\end{equation}

# Therefore:

\begin{equation}
p(target=1 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)) = \frac{p((target=1) \cap (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))}{p((1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))}
\end{equation}

In [31]:
boolean_mask_5 = df[(df['target']==1) & (df['var_0']>= 1) & (df['var_0']<= 15) & (df['var_1']>= -10) & (df['var_1']<= 9.5) & (df['var_2']>=4 & (df['var_2']<=18))]

In [32]:
boolean_mask_5

ID_code  target    var_0   var_1    var_2    var_3    var_4  \
29          train_29       1   5.3301 -2.6064  13.1913   3.1193   6.6483   
63          train_63       1   7.7072  0.0183   9.9974   8.3524   9.2886   
65          train_65       1  10.5358 -2.5439   8.7394   6.7548  14.4099   
71          train_71       1   6.7547  2.5973  14.2141   8.3514   7.4942   
72          train_72       1   8.6470  0.4462  10.9808  10.1859  13.9121   
...              ...     ...      ...     ...      ...      ...      ...   
199966  train_199966       1  13.5797  2.5526   6.0512   5.2730  12.2182   
199976  train_199976       1   7.9663 -2.8485   9.0919   7.3298   9.6690   
199981  train_199981       1  12.8140  0.6386  14.1657   7.1044   8.9365   
199986  train_199986       1  12.0298 -8.7800   7.7071   7.4015   9.2305   
199990  train_199990       1  14.1475  1.8568  11.0066   3.6779  12.1944   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
29      -6.5659  5.9064  15.2341  ...   5.9215   7.9676   2.3405   1.1482   
63     -13.3627  6.0425  10.1108  ...   1.5832   5.0039   3.8814   7.4241   
65      -3.8724  5.1584  15.8381  ...   4.6648   6.4227   3.4025  -4.0882   
71      -1.3055  4.2336  15.0243  ...   1.4677   3.5935   2.0013   1.5777   
72     -16.4645  5.2837  13.8396  ...   6.6004  13.0969   2.2437   0.6260   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199966  -3.4048  7.3623  17.8372  ...   3.9023   7.9986   0.5213   2.3442   
199976 -16.7872  4.5094  12.4351  ...  10.7930   4.3876   3.7257   7.7038   
199981  -0.3274  6.5949  14.6078  ...   4.8208   4.9010   2.2513   0.7308   
199986 -16.2174  5.9064  17.9268  ...  16.7461   3.1249  -0.3943   8.4059   
199990 -16.5936  5.3217  14.8508  ...  -6.4708   4.7287   1.9034   7.2324   

        var_194  var_195  var_196  var_197  var_198  var_199  
29      23.2168  -2.0105   3.7600   9.4513  17.4105 -14.6897  
63      21.4844  -0.8297  -3.0468   7.5790  15.7685   5.4769  
65      14.1174  -0.2472   5.3847   8.6949  15.1340   3.8449  
71      18.2820  -4.3408   6.8869   9.3567  18.9013  13.3447  
72      23.4572   0.3117   8.1634   7.9248  10.5462 -17.8708  
...         ...      ...      ...      ...      ...      ...  
199966  14.5510  -1.1530   8.9883   8.3389   9.5440   4.2493  
199976  14.7384   0.1561   1.5794   8.4627  14.3604  -1.6688  
199981  14.7155   1.1464   5.5158   8.6519  16.0341   7.3809  
199986  14.3367   3.0991   4.3853   8.8019  15.0031  -0.3659  
199990  20.6047   1.7170  -4.0032   9.1627  13.8077  -1.9646  

[17299 rows x 202 columns]

In [33]:
joint_probablity_5 = boolean_mask_5.shape[0]/ df.shape[0]

In [34]:
joint_probablity_5

0.086495

In [35]:
boolean_mask_6 = df[(df['var_0']>= 1) & (df['var_0']<= 15) & (df['var_1']>= -10) & (df['var_1']<= 9.5) & (df['var_2']>=4 & (df['var_2']<=18))]

In [36]:
boolean_mask_6

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[181435 rows x 202 columns]

In [37]:
joint_probablity_6 = boolean_mask_6.shape[0]/df.shape[0]

In [38]:
joint_probablity_6

0.907175

In [39]:
conditional_probablity_3 = joint_probablity_5/joint_probablity_6

In [40]:
conditional_probablity_3

0.09534544051588724

# so the conditional probablity for equation:

\begin{equation}
p(target=1 | (1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18))= 0.095345
\end{equation} 

# Now lets compute the probablity for equation:

\begin{equation}
p((1 \leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) | target=1)
\end{equation}

# lets take event (A) = ((1<= Var_0 <= 15) & (-10<= Var_1 <= 9.5) & (4<= Var_2 <= 18)) and event (B) = (target =1)


\begin{equation}
p((1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) | target = 1) = \frac{p(1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)\cap target = 1)}{p(target = 1)}
\end{equation}

# we have to compute two probablities again:

\begin{equation}
p((1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18) \cap (target = 1)
\end{equation}

# and 

\begin{equation}
p(target = 1)
\end{equation}


In [41]:
boolean_mask_7 = df[(df['var_0']>= 1) & (df['var_0']<= 15) & (df['var_1']>= -10) & (df['var_1']<= 9.5) & (df['var_2'] >= 4) & (df['var_2']<=18) & (df['target']== 1)]

In [42]:
boolean_mask_7

ID_code  target    var_0   var_1    var_2    var_3    var_4  \
29          train_29       1   5.3301 -2.6064  13.1913   3.1193   6.6483   
63          train_63       1   7.7072  0.0183   9.9974   8.3524   9.2886   
65          train_65       1  10.5358 -2.5439   8.7394   6.7548  14.4099   
71          train_71       1   6.7547  2.5973  14.2141   8.3514   7.4942   
72          train_72       1   8.6470  0.4462  10.9808  10.1859  13.9121   
...              ...     ...      ...     ...      ...      ...      ...   
199966  train_199966       1  13.5797  2.5526   6.0512   5.2730  12.2182   
199976  train_199976       1   7.9663 -2.8485   9.0919   7.3298   9.6690   
199981  train_199981       1  12.8140  0.6386  14.1657   7.1044   8.9365   
199986  train_199986       1  12.0298 -8.7800   7.7071   7.4015   9.2305   
199990  train_199990       1  14.1475  1.8568  11.0066   3.6779  12.1944   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
29      -6.5659  5.9064  15.2341  ...   5.9215   7.9676   2.3405   1.1482   
63     -13.3627  6.0425  10.1108  ...   1.5832   5.0039   3.8814   7.4241   
65      -3.8724  5.1584  15.8381  ...   4.6648   6.4227   3.4025  -4.0882   
71      -1.3055  4.2336  15.0243  ...   1.4677   3.5935   2.0013   1.5777   
72     -16.4645  5.2837  13.8396  ...   6.6004  13.0969   2.2437   0.6260   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199966  -3.4048  7.3623  17.8372  ...   3.9023   7.9986   0.5213   2.3442   
199976 -16.7872  4.5094  12.4351  ...  10.7930   4.3876   3.7257   7.7038   
199981  -0.3274  6.5949  14.6078  ...   4.8208   4.9010   2.2513   0.7308   
199986 -16.2174  5.9064  17.9268  ...  16.7461   3.1249  -0.3943   8.4059   
199990 -16.5936  5.3217  14.8508  ...  -6.4708   4.7287   1.9034   7.2324   

        var_194  var_195  var_196  var_197  var_198  var_199  
29      23.2168  -2.0105   3.7600   9.4513  17.4105 -14.6897  
63      21.4844  -0.8297  -3.0468   7.5790  15.7685   5.4769  
65      14.1174  -0.2472   5.3847   8.6949  15.1340   3.8449  
71      18.2820  -4.3408   6.8869   9.3567  18.9013  13.3447  
72      23.4572   0.3117   8.1634   7.9248  10.5462 -17.8708  
...         ...      ...      ...      ...      ...      ...  
199966  14.5510  -1.1530   8.9883   8.3389   9.5440   4.2493  
199976  14.7384   0.1561   1.5794   8.4627  14.3604  -1.6688  
199981  14.7155   1.1464   5.5158   8.6519  16.0341   7.3809  
199986  14.3367   3.0991   4.3853   8.8019  15.0031  -0.3659  
199990  20.6047   1.7170  -4.0032   9.1627  13.8077  -1.9646  

[17270 rows x 202 columns]

In [43]:
joint_probablity_7 = boolean_mask_7.shape[0]/df.shape[0]

In [44]:
joint_probablity_7

0.08635

In [45]:
boolean_mask_8 = df[df['target']== 1]

In [46]:
boolean_mask_8

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
13          train_13       1  16.3699  1.5934  16.7395  7.3330  12.1450   
29          train_29       1   5.3301 -2.6064  13.1913  3.1193   6.6483   
63          train_63       1   7.7072  0.0183   9.9974  8.3524   9.2886   
65          train_65       1  10.5358 -2.5439   8.7394  6.7548  14.4099   
71          train_71       1   6.7547  2.5973  14.2141  8.3514   7.4942   
...              ...     ...      ...     ...      ...     ...      ...   
199966  train_199966       1  13.5797  2.5526   6.0512  5.2730  12.2182   
199976  train_199976       1   7.9663 -2.8485   9.0919  7.3298   9.6690   
199981  train_199981       1  12.8140  0.6386  14.1657  7.1044   8.9365   
199986  train_199986       1  12.0298 -8.7800   7.7071  7.4015   9.2305   
199990  train_199990       1  14.1475  1.8568  11.0066  3.6779  12.1944   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
13       5.9004  4.8222  20.9729  ...   7.4002   7.4031   4.3989   4.0978   
29      -6.5659  5.9064  15.2341  ...   5.9215   7.9676   2.3405   1.1482   
63     -13.3627  6.0425  10.1108  ...   1.5832   5.0039   3.8814   7.4241   
65      -3.8724  5.1584  15.8381  ...   4.6648   6.4227   3.4025  -4.0882   
71      -1.3055  4.2336  15.0243  ...   1.4677   3.5935   2.0013   1.5777   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199966  -3.4048  7.3623  17.8372  ...   3.9023   7.9986   0.5213   2.3442   
199976 -16.7872  4.5094  12.4351  ...  10.7930   4.3876   3.7257   7.7038   
199981  -0.3274  6.5949  14.6078  ...   4.8208   4.9010   2.2513   0.7308   
199986 -16.2174  5.9064  17.9268  ...  16.7461   3.1249  -0.3943   8.4059   
199990 -16.5936  5.3217  14.8508  ...  -6.4708   4.7287   1.9034   7.2324   

        var_194  var_195  var_196  var_197  var_198  var_199  
13      17.3638  -1.3022   9.6846   9.0419  15.6064 -10.8529  
29      23.2168  -2.0105   3.7600   9.4513  17.4105 -14.6897  
63      21.4844  -0.8297  -3.0468   7.5790  15.7685   5.4769  
65      14.1174  -0.2472   5.3847   8.6949  15.1340   3.8449  
71      18.2820  -4.3408   6.8869   9.3567  18.9013  13.3447  
...         ...      ...      ...      ...      ...      ...  
199966  14.5510  -1.1530   8.9883   8.3389   9.5440   4.2493  
199976  14.7384   0.1561   1.5794   8.4627  14.3604  -1.6688  
199981  14.7155   1.1464   5.5158   8.6519  16.0341   7.3809  
199986  14.3367   3.0991   4.3853   8.8019  15.0031  -0.3659  
199990  20.6047   1.7170  -4.0032   9.1627  13.8077  -1.9646  

[20098 rows x 202 columns]

In [47]:
joint_probablity_8 = boolean_mask_8.shape[0]/ df.shape[0]

In [48]:
joint_probablity_8

0.10049

In [49]:
conditional_probablity_4 = joint_probablity_7/joint_probablity_8

In [50]:
conditional_probablity_4

0.8592894815404518

# so the conditional probablity for the equation:

\begin{equation}
\frac{p(1\leqslant var_0 \leqslant 15)\cap (-10 \leqslant var_1 \leqslant 9.5)\cap (4 \leqslant var_2 \leqslant 18)\cap target = 1)}{p(target = 1)} = 0.85928
\end{equation}
